src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -1126,19 +1126,30 @@ public AssemblyFeatureInfo(EcmaModule module, Logger logger, IReadOnlyDictionary
                 Module = module;
                 RemovedAttributes = new HashSet<TypeDesc>();
 
-                PEMemoryBlock resourceDirectory = module.PEReader.GetSectionData(module.PEReader.PEHeaders.CorHeader.ResourcesDirectory.RelativeVirtualAddress);
+                // System.Private.CorLib has a special functionality that could delete an attribute in all modules.
+                // In order to get the set of attributes that need to be removed the modules need collect both the
+                // set of attributes in it's embedded XML file and the set inside System.Private.CorLib embedded
+                // XML file
+                ParseLinkAttributesXml(module, logger, featureSwitchValues, globalAttributeRemoval: false);
+                ParseLinkAttributesXml(module, logger, featureSwitchValues, globalAttributeRemoval: true);
+            }
+
+            public void  ParseLinkAttributesXml(EcmaModule module, Logger logger, IReadOnlyDictionary<string, bool> featureSwitchValues, bool globalAttributeRemoval)
+            {
+                EcmaModule xmlModule = globalAttributeRemoval ? (EcmaModule) module.Context.SystemModule : module;
```

Comment:
I rarely have to comment on formatting but the IL Linker formatting rules appearing where they shouldn't is starting to be a nuisance (this comment is not directly on your PR - I've commented on these in other PRs of people who work in both codebases in the past weeks, cc @vitek-karas).

There's an undeniable overhead in switching between styles. This is starting to be a friction point in PRs. Formatting never was a friction point. Honestly the linker coding style is creating a problem that I haven't seen in my 10 years on the team.

```suggestion
                EcmaModule xmlModule = globalAttributeRemoval ? (EcmaModule)module.Context.SystemModule : module;
```

Reply:
We should switch to linker in runtime repo real soon (hopefully this week). Once that is done I think the first thing we do on the linker is to update the formatting to match runtime rules. And then all of this will be history...

Reply:
> the first thing we do on the linker is to update the formatting to match runtime rules

Before doing that, it might be good to catch up parts of NativeAOT that are line-by-line ports of ILLink and delete ReferenceSource. Reformatting will introduce lots of diffs to ReferenceSource. It's not impossible to do the catchup later, but it will be an extra annoyance. Once it's all caught up, we don't need ReferenceSource anymore - we just have to be diligent in PRs to update both files always.